pkg/controllers/cluster/cluster_controller.go

```diff
@@ -321,6 +329,38 @@ func (c *Controller) ExecutionSpaceExistForCluster(clusterName string) (bool, er
 	return true, nil
 }
 
+// Delete finalizers of work objects
+func (c *Controller) deleteFinalizerForWorks(cluster *clusterv1alpha1.Cluster, workSpace *corev1.Namespace) error {
+	workList := &workv1alpha1.WorkList{}
+	err := c.Client.List(context.TODO(), workList, &client.ListOptions{
+		Namespace: workSpace.Name,
+	})
+	if err != nil {
+		klog.Errorf("Failed to list works in %s: %s", workSpace.Name, err)
+		return err
+	}
+	for i := range workList.Items {
```

Comment:
There is no need to return immediately upon encountering an error, as it may only be a temporary situation where a single work cannot be removed from `Finalizer`, without affecting the removal of other works.

How about this?

``` go
       var  errors []error
	for i := range workList.Items {
		work := &workList.Items[i]
		err = c.removeWorkFinalizer(work)
		if err != nil {
			errors = append(errors,fmt.Errorf("Error while removing finalizers of works %s: %w", work.Name, err))
		}
	}

    return utilerrors.NewAggregate(errors)

```

Reply:
fixed